src/sentry/tasks/groupowner.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import, print_function
+
+import logging
+
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.eventstore.models import Event
+from sentry.eventstore.processing import event_processing_store
+from sentry.models import Project
+from sentry.models.groupowner import GroupOwner, GroupOwnerType
+from sentry.tasks.base import instrumented_task
+from sentry.utils.committers import get_serialized_event_file_committers
+
+PREFERRED_GROUP_OWNERS = 3
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=1)
+
+logger = logging.getLogger("tasks.groupowner")
+
+
+@instrumented_task(
+    name="sentry.tasks.process_suspect_commits", default_retry_delay=5, max_retries=5
+)
+def process_suspect_commits(group_id, cache_key, **kwargs):
+    if group_id:
```

Comment:
If you have the cleanup branch in `post_process_group` for groupless events you could return early on the negation 

```python
if not group_id:
    return
```
You could also do the cleanup here for groupless events.

```python
if not group_id:
    event_processing_store.delete_by_key(cache_key)
    return
```

That would save you a layer of indentation for the entire function.

Reply:
Thanks Mark. I really prefer your suggestion over what I did...been a couple times now that I branched this way and had it pointed out. Gotta be more mindful of this pattern.